XamarinCommunityToolkit/Effects/SafeAreaEffectRouter.ios.cs

```diff
@@ -14,34 +14,39 @@ public class SafeAreaEffectRouter : PlatformEffect
 	{
 		Thickness initialMargin;
 
+		new View Element
+			=> base.Element as View;
+
+		bool IsEligibleToConsumeEffect
+			=> Element != null
+				&& UIDevice.CurrentDevice.CheckSystemVersion(11, 0)
+				&& UIApplication.SharedApplication.Windows.Any();
+
 		protected override void OnAttached()
 		{
-			if (!(Element is View element))
-				return;
-
-			if (!UIDevice.CurrentDevice.CheckSystemVersion(11, 0) || !UIApplication.SharedApplication.Windows.Any())
+			if (!IsEligibleToConsumeEffect)
 				return;
 
 			var insets = UIApplication.SharedApplication.Windows[0].SafeAreaInsets;
-
-			initialMargin = element.Margin;
-
-			if (insets.Top <= 0)
-				return;
-
-			var safeArea = SafeAreaEffect.GetSafeArea(element);
-
-			element.Margin = new Thickness(
-				initialMargin.Left + (safeArea.Left ? insets.Left : 0),
-				initialMargin.Top + (safeArea.Top ? insets.Top : 0),
-				initialMargin.Right + (safeArea.Right ? insets.Right : 0),
-				initialMargin.Bottom + (safeArea.Bottom ? insets.Bottom : 0));
+			var safeArea = SafeAreaEffect.GetSafeArea(Element);
+
+			initialMargin = Element.Margin;
+			Element.Margin = new Thickness(
+				initialMargin.Left + CalculateInsets(insets.Left, safeArea.Left),
+				initialMargin.Top + CalculateInsets(insets.Top, safeArea.Top),
+				initialMargin.Right + CalculateInsets(insets.Right, safeArea.Right),
+				initialMargin.Bottom + CalculateInsets(insets.Bottom, safeArea.Bottom));
 		}
 
 		protected override void OnDetached()
 		{
-			if (Element is Layout element)
```

Comment:
Technically this check has changed now since we are not checking anymore if the Element is a Layout? Do I see that correctly?

Reply:
Yes, previously we checked if Element is a subclass of **View** on attaching and for some reason checked that the Element is a subclass of **Layout** on detaching. I guess it was a mistake